### PR TITLE
Correct description about math.random algorithm

### PR DESCRIPTION
**Pull request by Someon1e**
## Changes

Corrected description about math.random algorithm.

## Checks

By submitting your pull request for review, you agree to the following:

- [x] This contribution was created in whole or in part by me, and I have the right to submit it under the terms of this repository's open source licenses.
- [x] I understand and agree that this contribution and a record of it are public, maintained indefinitely, and may be redistributed under the terms of this repository's open source licenses.
- [x] To the best of my knowledge, all proposed changes are accurate.

### CODE REVIEW

**Comment by IgnisRBX** [comment]
Hi @Someon1e, does this citation come from an official announcement, release note, or other source-of-truth?

**Comment by Someon1e** [comment]
> Hi @Someon1e, does this citation come from an official announcement, release note, or other source-of-truth?

The information is available on the [Luau website](https://luau-lang.org/compatibility), in the [Luau source code](https://github.com/luau-lang/luau/blob/43bf7c4e051b0d49dbb2bd3cbb2471d235da55db/VM/src/lmathlib.cpp#L16), and I also referenced the [PCG website](https://www.pcg-random.org/).

File: content/en-us/reference/engine/libraries/math.yaml [hunk 1]
```diff
@@ -465,6 +461,8 @@ functions:
 
       When called with two integer numbers `m` and `n`, returns a uniform
       pseudo-random integer in the range of `m` to `n`, inclusive.
+
+      Internally, this uses a 32-bit PCG (Permuted Congruential Generator), which achieves excellent statistical performance. However, it is not suitable for cryptographic applications.
```
[the remark this answers is not in the document]
Comment (IgnisRBX) [reply content/en-us/reference/engine/libraries/math.yaml:465]: Can we go with this one? Just to be slightly less prescriptive about what it should or shouldn't be used for...

**Review by IgnisRBX — APPROVED** [review]

**Comment by Jayden234137** [comment]
Fun

**Pull request merged**